web/src/components/tasks/passenger-transport-cdl/PassengerTransportCdl.tsx

```diff
@@ -0,0 +1,108 @@
+import { Task } from "@/lib/types/types";
+import React, { ReactElement } from "react";
+import { TaskHeader } from "@/components/TaskHeader";
+import {
+  createDataFormErrorMap,
+  DataFormErrorMapContext,
+} from "@/contexts/dataFormErrorMapContext";
+import { useFormContextHelper } from "@/lib/data-hooks/useFormContextHelper";
+import { PassengerTransportCdlQuestion } from "@/components/tasks/passenger-transport-cdl/PassengerTransportCdlQuestion";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { isBoolean } from "lodash";
+import { PassengerTransportCdlResult } from "@/components/tasks/passenger-transport-cdl/PassengerTransportCdlResult";
+import { Alert } from "@/components/njwds-extended/Alert";
+import { Content } from "@/components/Content";
+
+interface Props {
+  task: Task;
+  CMS_ONLY_show_error?: boolean;
+  CMS_ONLY_show_tab_2?: boolean;
+  CMS_ONLY_school_bus_radio_value?: boolean;
+  CMS_ONLY_passengers_radio_value?: boolean;
+}
+
+export const PassengerTransportCdl = (props: Props): ReactElement => {
+  const { Config } = useConfig();
+  const { business } = useUserData();
+  const [schoolBusRadioValue, setSchoolBusRadioValue] = React.useState<boolean | undefined>(
+    business?.roadmapTaskData?.passengerTransportSchoolBus,
+  );
+  const [passengersRadioValue, setPassengersRadioValue] = React.useState<boolean | undefined>(
+    business?.roadmapTaskData?.passengerTransportSixteenOrMorePassengers,
+  );
+
+  const {
+    FormFuncWrapper,
+    getInvalidFieldIds,
+    onSubmit,
+    state: formContextState,
+  } = useFormContextHelper(createDataFormErrorMap());
+
+  const showAlert = props.CMS_ONLY_show_error || getInvalidFieldIds().length > 0;
+  const bothInputsAnswered =
+    isBoolean(business?.roadmapTaskData?.passengerTransportSchoolBus) &&
+    isBoolean(business?.roadmapTaskData?.passengerTransportSixteenOrMorePassengers);
+
+  const showResults = props.CMS_ONLY_show_tab_2 || bothInputsAnswered;
+
+  const fieldnameToConfig: Record<string, string> = {
+    passengerTransportSchoolBus: Config.passengerTransportCdlTabOne.firstQuestionAlertFieldNameText,
+    passengerTransportSixteenOrMorePassengers:
+      Config.passengerTransportCdlTabOne.secondQuestionAlertFieldNameText,
+  };
+
+  return (
+    <>
+      <DataFormErrorMapContext.Provider value={formContextState}>
+        <div className="flex1 flex-column1 space-between1 min-height-38rem">
+          <TaskHeader
+            task={props.task}
+            tooltipText={Config.passengerTransportCdlTabOne.disabledTooltipText}
+          />
+          {showAlert && (
+            <Alert variant="error">
+              <Content>{Config.passengerTransportCdlTabOne.pageLevelAlertText}</Content>
+              {props.CMS_ONLY_show_error &&
```

Comment:
Added logic to render the list of field names within the error alert. Since these field names can change dynamically based on content, this applies only to the CMS.